components/eamxx/src/share/property_checks/mass_and_energy_column_conservation_check.cpp

```diff
@@ -278,6 +283,152 @@ PropertyCheck::ResultAndMsg MassAndEnergyColumnConservationCheck::check() const
+  const auto nlevs = m_num_levs;
+
+  //keep dt for fixers with fluxes.
+  //we do not plan to use fluxes in dycore fixer, but the code is already there
+  EKAT_REQUIRE_MSG(!std::isnan(m_dt), "Error! Timestep dt must be set in MassAndEnergyConservationCheck "
+                                      "before running check().");
+  auto dt = m_dt;
+
+  const auto pseudo_density = m_fields.at("pseudo_density").get_view<const Real**> ();
+  const auto T_mid          = m_fields.at("T_mid"         ).get_view<      Real**> ();
+  const auto horiz_winds    = m_fields.at("horiz_winds"   ).get_view<const Real***>();
+  const auto qv             = m_fields.at("qv"            ).get_view<const Real**> ();
+  const auto qc             = m_fields.at("qc"            ).get_view<const Real**> ();
+  const auto qi             = m_fields.at("qi"            ).get_view<const Real**> ();
+  const auto qr             = m_fields.at("qr"            ).get_view<const Real**> ();
+  const auto ps             = m_fields.at("ps"            ).get_view<const Real*>  ();
+  const auto phis           = m_fields.at("phis"          ).get_view<const Real*>  ();
+
+  const auto vapor_flux = m_fields.at("vapor_flux").get_view<const Real*>();
+  const auto water_flux = m_fields.at("water_flux").get_view<const Real*>();
+  const auto ice_flux   = m_fields.at("ice_flux"  ).get_view<const Real*>();
+  const auto heat_flux  = m_fields.at("heat_flux" ).get_view<const Real*>();
+
+  const auto policy = ExeSpaceUtils::get_default_team_policy(ncols, nlevs);
+
+  using namespace ekat::units;
+  using namespace ShortFieldTagsNames;
+
+  //better to get layouts from m_grid, even scalar0d
+  FieldLayout scalar0d({},{});
+  FieldIdentifier s0_fid("s0", scalar0d, kg / kg, m_grid->name());
+  Field res(s0_fid);
+  res.allocate_view();
+  //this does not exist, why? cause it can be a LEV or a COL array?
+  //FieldLayout scalar1d = m_grid->get_1d_scalar_layout();
+  FieldLayout scalar1d{{COL}, {m_num_cols}};
+  FieldIdentifier s1_fid("s1", scalar1d, kg / kg, m_grid->name());
+
+  Field field_version_s1(s1_fid);
+  field_version_s1.allocate_view();
+  auto field_view_s1 = field_version_s1.get_view<Real*>();
+
+  auto area = m_grid->get_geometry_data("area").clone();
+  auto area_view = area.get_view<const Real*>();
+
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA (const KT::MemberType& team) {
+    const int i = team.league_rank();
+    const auto pseudo_density_i = ekat::subview(pseudo_density, i);
+    // Calculate total gas mass (sum dp, no water loading)
+    field_view_s1(i) = compute_gas_mass_on_column(team, nlevs, pseudo_density_i) * area_view(i);
+  });
+
+  //reprosum vars
+  const Real* send;
+  Real* recv;
+  Int nlocal = ncols;
+  Int ncount = 1;
+  auto s1_host = field_version_s1.get_view<const Real*,Host>();
+  auto s0_host = res.get_view<Real,Host>();
+  send = s1_host.data();
+  recv = s0_host.data();
+
+  eamxx_repro_sum(send, recv, nlocal, ncount, MPI_Comm_c2f(m_comm.mpi_comm()));
```

Comment:
You could do just one global reduction by combining this call to repro_sum with the next.